bolt-simple-taproot.md

```diff
@@ -0,0 +1,1141 @@
+
+- MUST fail the channel if `next_local_nonce` is absent.
+
+- If the local nonce generation is non-deterministic and the recipient co-signs
+  commitments only upon pending broadcast:
+
+  - MUST **securely** store the local nonce.
+
+### Message Retransmission
+
+#### `channel_reestablish` Extensions
+
+We add a new TLV field to the `channel_reestablish` message:
+
+1. `tlv_stream`: `channel_reestablish_tlvs`
+2. types:
+1. type: 4 (`next_local_nonce`)
+2. data:
+   * [`66*byte`: `public_nonce`]
+
+Similar to the `next_per_commitment_point`, by sending the `next_local_nonce`
+value in this message, we ensure that the remote party has our public nonce
+which is required to generate a new commitment signature.
+
+##### Requirements
+
+The sender:
+
+- MUST set `next_local_nonce` to a fresh, unique `musig2` nonce as specified by
+  `bip-musig2`
+
+The recipient:
+
+- MUST fail the channel if `next_local_nonce` is absent.
+
+A node:
+
+  - If ALL of the following conditions apply:
+  
+    - It has previously sent a `commitment_signed` message
+  
+    - It never processed the corresponding `revoke_and_ack` message
+  
+    - It decides to retransmit the exact `update_` messages from the last sent
+      `commitment_signed`
+  
+  - THEN it must regenerate the partial signature using the newly received
+    `next_local_nonce`
+
+### Funding Transactions
+
+For our Simple Taproot Channels, `musig2` is used to generate a single
+aggregated public key.
+
+The resulting funding output script takes the form:
+`OP_1 funding_key`
+
+where:
+
+  * `funding_key = combined_funding_key + tagged_hash("TapTweak", combined_funding_key)*G`
+  * `combined_funding_key = musig2.KeyAgg(musig2.KeySort(pubkey1, pubkey2))`
+
+The funding key is derived via the recommendation in BIP 341 that states: "If
+the spending conditions do not require a script path, the output key should
+commit to an unspendable script path instead of having no script path. This can
+be achieved by computing the output key point as `Q = P +
+int(hashTapTweak(bytes(P)))G`". We refer to BIP 86 for the specifics of this
+derivation.
+
+By using the musig2 sorting routine, we inherit the lexicographical key
+ordering from the base segwit channels. Throughout the lifetime of a channel
+any keys aggregated via musig2 also are assumed to use the `KeySort` routine
+
+### Commitment Transactions
+
+#### To Local Outputs
+
+For the simple taproot commitment type, we use the same flow of revocation or
+delay, but instead left the revocation case into the taproot key-spend case.
+
+The new output has the following form:
+
+  * `OP_1 to_local_output_key`
+  * where:
+    * `to_local_output_key = revocationpubkey + tagged_hash("TapTweak", revocationpubkey || to_delay_script_root)`
+    * `to_delay_script_root = tapscript_root([to_delay_script])`
+    * `to_delay_script` is the delay script:
+        ```
+        <local_delayedpubkey> OP_CHECKSIG
+        <to_self_delay> OP_CHECKSEQUENCEVERIFY OP_DROP
```

Comment:
[the shortest form possible](https://github.com/Roasbeef/lightning-rfc/pull/1#discussion_r886414660) has 3 different scripts (0=no checksequenceverify, 1=use the 1 from checksig like in the to_remote script, else the script that was before my pr in the bolt proposal), but this would get too complex.
using this one is possible and saves a byte, but keep in mind the counterparty decides your csv_delay (could be 0), and an unspendable script would be a too high risk (accepting a 0 local csv due to a bug or an oversight of the ln impl or some other tool)

Reply:
That might be shorter from the PoV of the individual scripts, but the added scripts increases the size of the control block (you can make one of them shorter though).

Reply:
i dont mean to include every script in the taptree, the node chooses (and the remote node enforces) that the right script is used
if csv_delay is 0: `<pubkey> OP_CHECKSIG`
if csv_delay is 1: `<pubkey> OP_CHECKSIG OP_CHECKSEQUENCEVERIFY`
else: `<pubkey> OP_CHECKSIGVERIFY <csv_delay> OP_CHECKSEQUENCEVERIFY`

Reply:
without being overly cute, switching to the 0/non-zero cases above 

e.g.
```
if csv_delay is 0: <pubkey> OP_CHECKSIG
else: <pubkey> OP_CHECKSIGVERIFY <csv_delay> OP_CHECKSEQUENCEVERIFY
```
makes things miniscript compatible and is trivial enough?